internal/core/api.rs

```diff
@@ -554,8 +554,24 @@ impl Window {
     ///
     /// Any position fields in the event must be in the logical pixel coordinate system relative to
     /// the top left corner of the window.
-    // TODO: Return a Result<(), PlatformError>
+    ///
+    /// This function panics if there is an error processing the event.
+    /// Use [`Self::try_dispatch_event()`] to handle the error.
+    #[track_caller]
```

Comment:
Perhaps we should just deprecate it?

Reply:
I was considering deprecating it, yes.

It's still used a bit.
 - A lot in our tests
 - in the Qt backend because Qt never returns error anyway
 - in the backends/winit/wasm_input_helper.rs because there is no way to return error from there
 - in the virtual_keyboard example
 - in the MinimalSoftwareWindow::set_size

(Would be nice is set_size and update_window_properties were returning a result but they don't)

Of course we could just replace everything with `try_dispatch_event(...).unwrap();` but that'd be a lot of churn for our tests.

I think it's fine to keep. It's like `borrow()`/`try_borrow()`

Reply:
👍 agreed and we can still do that later.